Source/SwiftLintFramework/Rules/ColonRule.swift

```diff
@@ -94,42 +94,90 @@ public struct ColonRule: CorrectableRule, ConfigurationProviderRule {
+        if configuration.applyToDictionaries {
+            dictionaryViolations = validateFile(file, dictionary: file.structure.dictionary)
+        } else {
+            dictionaryViolations = []
+        }
+
+        return (violations + dictionaryViolations).sorted { $0.location < $1.location }
     }
 
     public func correctFile(_ file: File) -> [Correction] {
-        let violations = violationRangesInFile(file, withPattern: pattern)
-        let matches = file.ruleEnabledViolatingRanges(violations, forRule: self)
+        let violations = correctionRangesInFile(file)
+        let matches = violations.filter {
+            !file.ruleEnabledViolatingRanges([$0.range], forRule: self).isEmpty
+        }
+
         guard !matches.isEmpty else { return [] }
         let regularExpression = regex(pattern)
         let description = type(of: self).description
         var corrections = [Correction]()
         var contents = file.contents
-        for range in matches.reversed() {
-            contents = regularExpression.stringByReplacingMatches(in: contents,
-                options: [], range: range, withTemplate: "$1: $2")
+        for (range, kind) in matches.reversed() {
+            switch kind {
+            case .type:
+                contents = regularExpression.stringByReplacingMatches(in: contents,
+                                                                      options: [],
+                                                                      range: range,
+                                                                      withTemplate: "$1: $2")
+            case .dictionary:
+                contents = contents.bridge().replacingCharacters(in: range, with: ": ")
+            }
+
             let location = Location(file: file, characterOffset: range.location)
             corrections.append(Correction(ruleDescription: description, location: location))
         }
         file.write(contents)
         return corrections
     }
 
-    // MARK: - Private
+    private typealias RangeWithKind = (range: NSRange, kind: ColonKind)
+
+    private func correctionRangesInFile(_ file: File) -> [RangeWithKind] {
+        let violations = violationRangesInFile(file, withPattern: pattern).map {
+            (range: $0, kind: ColonKind.type)
+        }
+        let dictionary = file.structure.dictionary
+        let contents = file.contents.bridge()
+        let dictViolations: [RangeWithKind] = dictionaryViolations(in: file,
+                                                                   dictionary: dictionary).flatMap {
+            guard let range = contents.byteRangeToNSRange(start: $0.location,
+                                                          length: $0.length) else {
+                return nil
+            }
+            return (range: range, kind: ColonKind.dictionary)
+        }
+
+        return (violations + dictViolations).sorted { $0.range.location < $1.range.location }
+    }
+}
+
+// MARK: Type Colon Rule
```

Comment:
elsewhere in SwiftLint's codebase, we tend to consistently have a single empty newline after `// MARK`s.